hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ClosePipelineSubcommand.java

```diff
@@ -35,13 +40,52 @@
     mixinStandardHelpOptions = true,
     versionProvider = HddsVersionProvider.class)
 public class ClosePipelineSubcommand extends ScmSubcommand {
+  @CommandLine.ArgGroup(multiplicity = "1")
+  private CloseOptionGroup closeOption;
 
-  @CommandLine.Parameters(description = "ID of the pipeline to close")
-  private String pipelineId;
+  @CommandLine.Mixin
+  private final FilterPipelineOptions filterOptions = new FilterPipelineOptions();
 
   @Override
   public void execute(ScmClient scmClient) throws IOException {
-    scmClient.closePipeline(
-        HddsProtos.PipelineID.newBuilder().setId(pipelineId).build());
+    if (!Strings.isNullOrEmpty(closeOption.pipelineId)) {
+      scmClient.closePipeline(HddsProtos.PipelineID.newBuilder().setId(closeOption.pipelineId).build());
+    }
+
+    if (closeOption.closeAll) {
+      Optional<Predicate<? super Pipeline>> replicationFilter = filterOptions.getReplicationFilter();
+
+      Stream<Pipeline> stream = scmClient.listPipelines()
+          .stream()
+          .filter(p -> p.getPipelineState() != Pipeline.PipelineState.CLOSED);
+      if (replicationFilter.isPresent()) {
+        stream = stream.filter(replicationFilter.get());
+      }
+
+      stream.forEach(pipeline -> {
+        try {
+          scmClient.closePipeline(
+              HddsProtos.PipelineID.newBuilder().setId(pipeline.getId().getId().toString()).build());
+        } catch (IOException e) {
+          System.err.println("Error closing pipeline: " + pipeline.getId() + ", cause: " + e.getMessage());
+        }
+      });
+    }
+  }
+
+  private static class CloseOptionGroup {
+    @CommandLine.Option(
+        names = {"pipelineId"},
```

Comment:
This should still be a `Parameter` (which is a value without a name).  Currently the command gives error:

```
$ ozone admin pipeline close 123
Unmatched argument at index 2: '123'
```